### PR TITLE
FactoryBean now sets the tablePrefix for taskBatchExecutionListener

### LINKED ISSUE #353: TaskBatchExecutionListener is not receiving tablePrefix
Class **TaskBatchExecutionListenerFactoryBean** is not receiving **tablePrefix** when it is created by **autoconfiguration** class **TaskBatchExecutionListenerAutoconfiguration**.
This causes TASK_TASK_**BATCH** table is always using default prefix even when property **spring.cloud.task.tablePrefix** is set.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #353

### CODE REVIEW

**Comment by mminella** [comment]
Unit test?

File: spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/PrefixTests.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+	private final static String DATASOURCE_DRIVER_CLASS_NAME = "org.h2.Driver";
+
+	private static int randomPort;
+
+	static {
+		randomPort = SocketUtils.findAvailableTcpPort();
+		DATASOURCE_URL = "jdbc:h2:tcp://localhost:" + randomPort + "/mem:dataflow;DB_CLOSE_DELAY=-1;"
+				+ "DB_CLOSE_ON_EXIT=FALSE";
+	}
```
Comment (mminella) [inline spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/PrefixTests.java]: Is any of this stuff needed?  I can clean it up on merge if you agree.

Comment (cppwfs) [reply spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/PrefixTests.java]: I agree.

**Comment by mminella** [comment]
Polished, rebase, and squashed as 01311e8be6c3a9d0560474f2c0b8ffd87cf393b4

**Pull request closed without merging**